lib/rules/color-no-out-gamut-range/index.js

```diff
@@ -0,0 +1,93 @@
+'use strict';
+
+const validateOptions = require('../../utils/validateOptions');
+const ruleMessages = require('../../utils/ruleMessages');
+const valueParser = require('postcss-value-parser');
+// @ts-expect-error -- TS7016
+const Color = require('colorjs.io').default;
+const report = require('../../utils/report');
+const getDeclarationValue = require('../../utils/getDeclarationValue');
+const { isAtRule } = require('../../utils/typeGuards');
+const declarationValueIndex = require('../../utils/declarationValueIndex');
+
+const ruleName = 'color-no-out-gamut-range';
+
+const messages = ruleMessages(ruleName, {
+	rejected: (color) => `Unexpected color out of gamut range "${color}"`,
+});
+
+const meta = {
+	url: 'https://stylelint.io/user-guide/rules/list/color-no-out-gamut-range',
+};
+
+/** @type {import('stylelint').Rule} */
+const rule = (primary) => {
+	return (root, result) => {
+		const validOptions = validateOptions(result, ruleName, {
+			actual: primary,
+		});
+
+		if (!validOptions) {
+			return;
+		}
+
+		root.walkDecls((decl) => {
+			const parsedValue = valueParser(getDeclarationValue(decl));
+
+			// ignore css variable declaration
+			if (decl.prop.startsWith('--')) return;
+
+			// ignore scss variable declaration
+			if (decl.prop.startsWith('$')) return;
+
+			parsedValue.walk((node) => {
+				if (node.type !== 'function') return;
+
+				if (
+					node.value !== 'lch' &&
+					node.value !== 'lab' &&
+					node.value !== 'oklch' &&
+					node.value !== 'oklab'
```

Comment:
[suggest] We can also use `Set` to reduce duplication slightly like this:
(feel free to ignore the suggestion)

```js
// Define the constant out of the `rule` function
const COLOR_FUNCTIONS = new Set(['lch', 'lab', 'oklch', 'oklab']);

// ...

if (!COLOR_FUNCTIONS.has(node.value)) return;

// ...
```

Reply:
Umm..., when I try the change, the test passes on my local machine. 🤔 

<details>
<summary>Diff</summary>

```diff
diff --git a/lib/rules/color-no-out-gamut-range/index.js b/lib/rules/color-no-out-gamut-range/index.js
index 86e6dff84..0d3ba836e 100644
--- a/lib/rules/color-no-out-gamut-range/index.js
+++ b/lib/rules/color-no-out-gamut-range/index.js
@@ -22,6 +22,8 @@ const meta = {
 	url: 'https://stylelint.io/user-guide/rules/list/color-no-out-gamut-range',
 };
 
+const COLOR_FUNCTIONS = new Set(['lch', 'lab', 'oklch', 'oklab']);
+
 /** @type {import('stylelint').Rule} */
 const rule = (primary) => {
 	return (root, result) => {
@@ -43,13 +45,7 @@ const rule = (primary) => {
 			parsedValue.walk((node, _index, nodes) => {
 				if (node.type !== 'function') return;
 
-				if (
-					node.value !== 'lch' &&
-					node.value !== 'lab' &&
-					node.value !== 'oklch' &&
-					node.value !== 'oklab'
-				)
-					return;
+				if (!COLOR_FUNCTIONS.has(node.value)) return;
 
 				const isInSrgbGamut = new Color(valueParser.stringify(nodes)).inGamut('srgb');
 
```

</details>